Assets/Scripts/Interaction Scripts/ControllerInput.cs

```diff
@@ -273,6 +301,32 @@ public Transform LeftUITransform()
 		//  RIGHT HAND GETTER METHODS  //
 		/*******************************/
 		
+		/// <summary>
+		/// Whether the right hand is currently touching an object.
+		/// </summary>
+        /// <returns>True if the right hand is currently touching an object.</returns>
+		public bool RightIsTouching()
+		{
+			return (RightController.GetComponent<VRTK_InteractTouch>().GetTouchedObject()) ? true : false;
+		}
+
+		/// <summary>
+		/// Whether the right hand is currently touching an object.
+		/// </summary>
+        /// <returns>True if the right hand is currently touching an object.</returns>
+		public bool RightIsTouchingWaypoint()
+		{
+			GameObject touched = RightController.GetComponent<VRTK_InteractTouch>().GetTouchedObject();
+			if (touched)
+			{
+				if (touched.tag == "waypoint")
+				{
+					return true;
+				}	
+			}
+			return false;
+		}
+
```

Comment:
Alternatively just keep the same kind of if statement/return/ternary statement throughout these very similar methods.